### PR TITLE
[SeaTunnel #767][common] Translate seatunnel-common scala code to java

### LINKED ISSUE #767: [Feature][common] Translate seatunnel-common scala code to java
### Search before asking

- [X] I had searched in the [feature](https://github.com/InterestingLab/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Translate seatunnel-common scala code to java.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by leo65535**
---

## Purpose of this pull request

Fixed #767

Translate seatunnel-common scala code to java

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by leo65535** [comment]
cc @wolfboys @yx91490

**Comment by wolfboys** [comment]
in Waterdrop.java line 138:

```
private static void baseCheckConfig(List<? extends Plugin>... plugins) {
        boolean configValid = true;
        for (List<? extends Plugin> pluginList : plugins) {
            for (Plugin plugin : pluginList) {
                CheckResult checkResult = null;
                try {
                    checkResult = plugin.checkConfig();
                } catch (Exception e) {
                    checkResult = new CheckResult(false, e.getMessage());
                }
                if (!checkResult.isSuccess()) {
                    configValid = false;
                    LOGGER.error("Plugin[{}] contains invalid config, error: {} \n", plugin.getClass().getName(), checkResult.getMsg());
                }
                if (!configValid) {
                    System.exit(-1); // invalid configuration
                }
            }
        }
        deployModeCheck();
    }

```

change to:

```
private static void baseCheckConfig(List<? extends Plugin>... plugins) {
    for (List<? extends Plugin> pluginList : plugins) {
        for (Plugin plugin : pluginList) {
            CheckResult checkResult = null;
            try {
                checkResult = plugin.checkConfig();
            } catch (Exception e) {
                checkResult = new CheckResult(false, e.getMessage());
            }
            if (!checkResult.isSuccess()) {
                LOGGER.error("Plugin[{}] contains invalid config, error: {} \n", plugin.getClass().getName(), checkResult.getMsg());
                System.exit(-1); // invalid configuration
            }
        }
    }
    deployModeCheck();
}

```

wouldn't it be better?

File: seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/config/Common.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.util.Optional;
+
+public class Common {
+
+    private static final List<String> ALLOWED_MODES = Arrays.asList("client", "cluster");
+
+    private static Optional<String> MODE = Optional.empty();
+
+    public static boolean isModeAllowed(String mode) {
+        return ALLOWED_MODES.stream().anyMatch(m -> mode.toLowerCase().equals(m));
```
Comment (wolfboys) [inline seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/config/Common.java:34]: return ALLOWED_MODES.stream().anyMatch(m -> mode.equalsIgnoreCase(m));

wouldn't it be better?

**Review by wolfboys — COMMENTED** [review]
LGTM, in some places, you can optimize the code

File: seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/RowConstant.java [hunk 2]
```diff
@@ -14,10 +14,11 @@
...
 
-object RowConstant {
-  val ROOT = "__root__"
-  val TMP = "__tmp__"
-  val JSON = "__json__"
-}
\ No newline at end of file
+package io.github.interestinglab.waterdrop.common;
+
+public class RowConstant {
```
Comment (yx91490) [inline seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/RowConstant.java:20]: how about change class to interface type? or you should add private constructor.

File: seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/config/Common.java (hunk 1 above)
Comment (yx91490) [reply seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/config/Common.java:34]: return ALLOWED_MODES.contains(mode.toLowerCase());
wouldn't it be better?

Comment (wolfboys) [reply seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/config/Common.java:34]: good job!

File: seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/config/Common.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            try {
+                String path = Common.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+                return Paths.get(path).getParent().getParent().getParent();
+            } catch (URISyntaxException e) {
+                throw new RuntimeException(e);
+            }
+        } else if (MODE.equals(Optional.of("cluster"))) {
+            return Paths.get("");
+        } else {
+            return null;
```
Comment (yx91490) [inline seatunnel-common/src/main/java/io/github/interestinglab/waterdrop/common/config/Common.java:71]: how about throw: 
throw new IllegalStateException("MODE not support : " + MODE.orElse("null"));

**Comment by leo65535** [comment]
Thanks for the suggestion, patch is updated accordingly. @wolfboys @yx91490

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**